### ISSUE TITLE
Font thinner or thicker depending on the client

### ISSUE THREAD

**Issue by aperezdc** [Igalia/wpewebkit.org#149]
In theory we are using a web font, so this should display practicvally in the same way, but I can consistently reproduce two different rendering styles.

GNOME Web (with WebKitGTK 2.38.4) and Firefox show the font slightly thicker in one of my machines:

![Thicker font style](https://user-images.githubusercontent.com/723451/219019285-8bb8c172-3049-4d32-b609-4adf993def8d.png)

OTOH, Chromium and Web show the font thinner in my other machine:

![Thinner font style](https://user-images.githubusercontent.com/723451/219019290-abd75d1f-9579-4ff8-a8fd-d220210f0645.png)

There is something funny going on here. The version of WebKitGTK in use is the _same_ for both computers, yet they show differently. I suspect something related to the list of installed fonts on each of them.

**Comment by meyerweb** [comment]
A good way to check is with the Fonts tab of the Firefox Web Inspector — that will tell you the font face and its metrics used for whatever element you’re inspecting.  If any of those things are different, we can figure out what to do.  If they’re all the same, it’s due to the engines’ (or machines’) font-rendering internals and out of our hands.

**Comment by aperezdc** [comment]
> A good way to check is with the Fonts tab of the Firefox Web Inspector — that will tell you the font face and its metrics used for whatever element you’re inspecting. If any of those things are different, we can figure out what to do. If they’re all the same, it’s due to the engines’ (or machines’) font-rendering internals and out of our hands.

No, it's not out of our hands. Note that I can get both renderings with the same engine (WebKitGTK 2.38.4) running on two different computers. Those two computers have also the same versions of the WebKitGTK dependencies as well... the main difference is the list of locally installed fonts.

On each machine I got a list with `fc-list -f '%{family}\n' | sort -u > fontlist` and the diffed the results. I am quite sure tha main difference is that the machine that has the thinner rendering does _not_ have the Source Sans font installed:

```diff
--- fclist-a	2023-02-15 15:30:25.039067642 +0200
+++ fclist-b	2023-02-15 15:30:43.755922829 +0200
...
@@ -349,18 +208,6 @@
 Source Code Pro,Source Code Pro Medium
 Source Code Pro,Source Code Pro Semibold
 Source Code Variable
-Source Sans 3
-Source Sans 3,Source Sans 3 Black
-Source Sans 3,Source Sans 3 ExtraLight
-Source Sans 3,Source Sans 3 Light
-Source Sans 3,Source Sans 3 Semibold
-Source Sans 3 VF
-Source Sans Pro
-Source Sans Pro,Source Sans Pro Black
-Source Sans Pro,Source Sans Pro ExtraLight
-Source Sans Pro,Source Sans Pro Light
-Source Sans Pro,Source Sans Pro Semibold
-Source Sans Variable
 Spline Sans
 Spline Sans Mono
 Spline Sans Mono,Spline Sans Mono Light
```

My hypothesis is that there is a variant which is intalled locally which is being picked instead of a webfont (which may be slightly different, or the webfont not even loaded). I'm looking into fixing this now because it needs to be possible to render the text similarly everywhere.

**Comment by aperezdc** [comment]
This seems to be an [issue in WebKit](https://bugs.webkit.org/show_bug.cgi?id=177039#c6), which is also dependant on whether the font is locally installed or not.

**Comment by aperezdc** [comment]
The WebKit fix is included in WebKitGTK and WPE 2.40.0, closing.
